### PR TITLE
[ASCII-2079] Add password log scrubber pattern

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

We only had a yaml matching rule for passwords but we should scrub loglines as well

### Describe how to test/QA your changes
* this is unit tested

### CODE REVIEW

File: pkg/util/scrubber/default.go [hunk 1]
```diff
@@ -94,11 +94,17 @@ func AddDefaultReplacers(scrubber *Scrubber) {
...
-	passwordReplacer := matchYAMLKeyPart(
+	yamlPasswordReplacer := matchYAMLKeyPart(
 		`(pass(word)?|pwd)`,
 		[]string{"pass", "pwd"},
 		[]byte(`$1 "********"`),
 	)
+	// replace all alphanum + special chars except quotes
+	passwordReplacer := Replacer{
+		// see test cases for match patterns
+		Regex: regexp.MustCompile(`(?i)(\"?(?:pass(?:word)?|pswd|pwd)\"?)( = |=|: \"?)([0-9A-Za-z#!$%&()*+,\-./:<=>?@[\\\]^_{|}~]+)`),
```
Comment (pgimalac) [inline pkg/util/scrubber/default.go:105]: What about `;` ? I know that could scrub too much (eg. in the SQL connection error), but better too much than not enough

File: pkg/util/scrubber/default_test.go [hunk 2]
```diff
@@ -295,6 +295,62 @@ func TestTextStripURLPassword(t *testing.T) {
 		`Connection dropped : ftp://user:********@host:port`)
 }
 
+func TestTextStripLogPassword(t *testing.T) {
```
Comment (pgimalac) [inline pkg/util/scrubber/default_test.go:298]: Can you add one case with the SQL connection error, similar to the one reported by support ?

File: pkg/util/scrubber/default.go (hunk 1 above)
Comment (pgimalac) [inline pkg/util/scrubber/default.go:105]: Why not just use <everything but space characters> for the password ?
```suggestion
		Regex: regexp.MustCompile(`(?i)(\"?(?:pass(?:word)?|pswd|pwd)\"?)( = |=|: \"?)([[:^space:]]+)`),
```

File: pkg/util/scrubber/default_test.go (hunk 2 above)
Comment (GustavoCaso) [reply pkg/util/scrubber/default_test.go:298]: Agree that having more test cases testing the password value it self would be useful 😄

File: pkg/util/scrubber/default.go (hunk 1 above)
Comment (GustavoCaso) [inline pkg/util/scrubber/default.go:105]: Wondering if we could simply the regex a bit. 

`(?i)(\"?(?:pass(?:word)?|pswd|pwd)\"?)([:=\s]\"?)([^\s]+)`

Replaced `( = |=|: \"?)` with `([:=\s]\"?)` That would `=` `:`, space and the optional `"`

Replaced `([0-9A-Za-z#!$%&()*+,\-./:<=>?@[\\\]^_{|}~]+)` with `([^\s]+)` matches anything that is not a space

Comment (GustavoCaso) [reply pkg/util/scrubber/default.go:105]: I'm not familiar with `:space` but should it be `[^:space:]`?  Also, do we need the extra brackets?

Comment (pgimalac) [reply pkg/util/scrubber/default.go:105]: https://pkg.go.dev/regexp/syntax

```
[[:^alpha:]]   negated ASCII character class
```

Comment (pgimalac) [reply pkg/util/scrubber/default.go:105]: We might as well use `[[:space:]]`, it includes all whitespace characters
https://pkg.go.dev/regexp/syntax
```
[[:space:]]    whitespace (== [\t\n\v\f\r ])
```

**Review by pgimalac — APPROVED** [review]
LGTM

File: pkg/util/scrubber/default.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pgimalac) [reply pkg/util/scrubber/default.go:105]: Discussed offline, that's on purpose

Comment (pgimalac) [reply pkg/util/scrubber/default.go:105]: Left as is to avoid scrubbing too much in SQL queries or URLs

Comment (pgimalac) [reply pkg/util/scrubber/default.go:105]: Excluding quotes and semi-colon is on purpose

**Pull request merged**